File: AI/ml/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 4 - Simple Linear Regression/SLR_MASTER.py
# Data Preprocessing SLR

##################################
### check code     : cmd + i   ###
### run code       : cmd enter ###
### set working dir: Fn F5     ###
##################################

# Import libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Import dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Split dataset into Training and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling (optional) necessary if there are Euclidian distance to be processed
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

# Fit SLR to the training set
from sklearn.linear_model import LinearRegression
# regressor (the 'machine') an object of the class
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predict the Test set results
y_pred = regressor.predict(X_test)
y_pred_train = regressor.predict(X_train)

# Visualise ***Training*** set results
plt.scatter(X_train, y_train, color='red')
plt.plot(X_train, y_pred_train, color='blue')
plt.title('Salary vs Experience (TRAINING set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

# Visualise ***Test*** set results
plt.scatter(X_test, y_test, color='red')
# NOTE: the lin regression line was already done using Training set
plt.plot(X_train, y_pred_train, color='green')
plt.title('Salary vs Experience (TEST set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: AI/ml/Machine Learning A-Z Template Folder/Part 2 - Regression/Section 4 - Simple Linear Regression/SLR_MASTER.R
# Data Preprocessing template

# Importing the dataset
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[, 2:3]

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling (optional) necessary if there are Euclidian distance to be processed
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

# Fitting Simple Linear Regression to Training set
regressor = lm(formula = Salary ~ YearsExperience, 
               data = training_set)
# Check info
summary(regressor)

# Predicting the Test set results, 
y_pred = predict(regressor, newdata = test_set)
y_pred

# Visualize Training set
# install.packages("ggplot2")
library(ggplot2)
ggplot() +
  geom_point(aes(x=training_set$YearsExperience, y=training_set$Salary),
             colour = "red") +
  geom_line(aes(training_set$YearsExperienc, y=predict(regressor, newdata = training_set)),
                colour = "blue") +
  ggtitle('Salary vs Experience (Training Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

# Visualize Test set
ggplot() +
  geom_point(aes(x=test_set$YearsExperience, y=test_set$Salary),
             colour = "red") +
  geom_line(aes(training_set$YearsExperienc, y=predict(regressor, newdata = training_set)),
            colour = "blue") +
  ggtitle('Salary vs Experience (Test Set)') +
  xlab('Years of Experience') +
  ylab('Salary')
            
            








